File: Free SysLog/Windows/Clear logs older than.vb
Public Class ClearLogsOlderThan
    Public dateChosenDate As Date

    Private Sub Clear_logs_older_than_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        DateTimePicker.MinDate = Date.Now.AddDays(-15)
        DateTimePicker.MaxDate = Date.Now
    End Sub

    Private Sub BtnClearLogs_Click(sender As Object, e As EventArgs) Handles BtnClearLogs.Click
        dateChosenDate = DateTimePicker.Value
        Close()
    End Sub

    Private Sub DateTimePicker_ValueChanged(sender As Object, e As EventArgs) Handles DateTimePicker.ValueChanged
        BtnClearLogs.Enabled = True
        DialogResult = DialogResult.OK
    End Sub

    Private Sub Clear_logs_older_than_KeyUp(sender As Object, e As KeyEventArgs) Handles Me.KeyUp
        If e.KeyCode = Keys.Escape Then
            DialogResult = DialogResult.Cancel
            Close()
        End If
    End Sub
End Class

File: Free SysLog/Windows/Clear logs older than.cs
using System;
using System.Windows.Forms;

namespace Free_SysLog
{
    public partial class ClearLogsOlderThan
    {
        public DateTime dateChosenDate;

        public ClearLogsOlderThan()
        {
            InitializeComponent();
        }

        private void Clear_logs_older_than_Load(object sender, EventArgs e)
        {
            DateTimePicker.MinDate = DateTime.Now.AddDays(-15);
            DateTimePicker.MaxDate = DateTime.Now;
        }

        private void BtnClearLogs_Click(object sender, EventArgs e)
        {
            dateChosenDate = DateTimePicker.Value;
            Close();
        }

        private void DateTimePicker_ValueChanged(object sender, EventArgs e)
        {
            BtnClearLogs.Enabled = true;
            DialogResult = DialogResult.OK;
        }

        private void Clear_logs_older_than_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
            {
                DialogResult = DialogResult.Cancel;
                Close();
            }
        }
    }
}